### ISSUE TITLE
 EpiCode.23Video.dll  contains Null Public Key Token

### ISSUE THREAD

**Issue by Yogesh2291** [dss-web/23-video-episerver#11]
After successful run of 23 video connector on local machine ,  when we trying to deploy on our testing environment then we found, only two Dlls (23API.dll and DotnetOpenAuth.dll) are deploying correctly. Third one EpiCode.23Video.dll is not going with deployment.

We analyzed the issue and found that both Dlls (23API.dll and DotnetOpenAuth.dll) have public key token value while the EpiCode.23Video.dll does not have any public key token. public key token is showing null value for EpiCode.23Video.dll.

<Reference Include="23API, Version=1.1.0.0, Culture=neutral, PublicKeyToken=c691d74182a532de, processorArchitecture=MSIL">
 
<Reference Include="DotNetOpenAuth, Version=3.4.5.10202, Culture=neutral, PublicKeyToken=2780ccd10d57b246, processorArchitecture=MSIL">
 
<Reference Include="EPiCode.23Video, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">  -> **does not contain public key token**
 
Could you please let us know the EpiCode.23video.dll is private assembly or does contain any public key token value?

As  EpiCode.23video.dll  contains null public key token , so we can not deploy it on other environment ? is it this reason behind this issue ?

Please help to understand

**Comment by tomahg** [comment]
So EpiCode.23Video.dll is omitted from the deployment? How are you running your deployment?

Can this be a solution?
https://stackoverflow.com/questions/2947017/reference-dll-file-not-copying-to-bin-with-deployment-project-causing-error

**Comment by Yogesh2291** [comment]
Hi Tom,

Thanks for the reply!

We already tried the steps given in the link you have shared, the two dll which have public key are deployed and available in the bin folder but the one without public key (EpiCode.23Video.dll) is not deployed when deployment is done.

I'm assuming issue is related to private dll, as the other two dll having public key is deployment and the one which is private is not deployed. or it related to something else?

**Comment by tomahg** [comment]
When you build the solution on your local computer, the dll ends up in your bin folder, right?

What kind of deployment are you doing, that makes the dll disappear? Azure DevOps Pipelines, TeamCity/Octopus deploy, xcopy...?

**Comment by Yogesh2291** [comment]
Hi Tomas,

We have one folder name ThirdParty. We kept all third party dlls in that folder. While using we gives reference of that folder to solution. All other third party dlls taking reference correctly while deploying but only one dll (EpiCode.23Video.dll) is not deploying.

We are using Azure Devops Pipelines for deployment.

Thanks
Yogesh K

**Comment by tomahg** [comment]
Do you have the **Copy Local = True** for EPiCode.23Video.dll?

**Comment by Yogesh2291** [comment]
Yeah we tried with Copy Local = True for EpiCode.23Video.dll but that did not work.

**Comment by suwayya** [comment]
Hi @Yogesh2291 
I've added signing to the 23Video project. This will hopefully fix your problem.

**Comment by Yogesh2291** [comment]
Hi Vidar,

I did not understand , what do you mean by signing in to 23video project.

Could you please explain it.

Thanks
Yogesh K,

**Comment by tomahg** [comment]
@Yogesh2291 : Vidar has updated the code, so that the EpiCode.23Video.dll will be *signed* and thus get a *public key token*.

Please try again.

Regards,
Tomas

**Comment by VidarMa** [comment]
Hei
I think you got the Wrong Vidar ☺

I'm not Vidar Olsen

Vidar Martinsen
Mobil: 928 53 659

Fra: Yogesh2291 <notifications@github.com>
Sendt: 18. november 2019 07.17
Til: dss-web/23-video-episerver <23-video-episerver@noreply.github.com>
Kopi: Subscribed <subscribed@noreply.github.com>
Emne: Re: [dss-web/23-video-episerver] EpiCode.23Video.dll contains Null Public Key Token (#11)

Hi Vidar,

I did not understand , what do you mean by signing in to 23video project.

Could you please explain it.

Thanks
Yogesh K,

**Comment by suwayya** [comment]
@Yogesh2291 visual studio should add the public token for you when you reference it now since the dll is signed after build. If you still have problems you can open "Developer Command Prompt for VS" and use this command:
`sn -Tp ./EPiCode.23Video.dll`
You will then get an output containing the public key and the public key token.
